homeassistant/components/osoenergy/sensor.py

```diff
@@ -0,0 +1,179 @@
+"""Support for OSO Energy sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.const import OSOEnergySensorData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfEnergy, UnitOfPower, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+import homeassistant.util.dt as dt_util
+
+from . import OSOEnergyEntity
+from .const import DOMAIN
+
+
+def _get_local_hour(utc_hour: int):
+    """Get the local hour."""
+    now = dt_util.utcnow()
+    now_local = dt_util.now()
+    utc_time = now.replace(hour=utc_hour, minute=0, second=0, microsecond=0)
+    local_hour = dt_util.as_local(utc_time)
+    return local_hour.replace(
+        year=now_local.year, month=now_local.month, day=now_local.day
+    )
+
+
+def _convert_profile_to_local(values):
+    """Convert UTC profile to local."""
+    profile = [None] * 24
+    for hour in range(24):
+        local_hour = _get_local_hour(hour)
+        profile[local_hour.hour] = values[hour]
+
+    return profile
+
+
+@dataclass(frozen=True, kw_only=True)
+class OSOEnergySensorEntityDescription(SensorEntityDescription):
+    """Class describing OSO Energy heater sensor entities."""
+
+    value: Callable[[OSOEnergy], StateType]
+
+
+SENSOR_TYPES: dict[str, OSOEnergySensorEntityDescription] = {
+    "profile": OSOEnergySensorEntityDescription(
+        key="profile",
+        translation_key="profile",
+        value=lambda device: _convert_profile_to_local(device.state),
+    ),
```

Comment:
Please share a screenshot + explanation of this sensor

Reply:
The profile sensor displays the current temperature profile of the device - 24 comma separated temperatures for the day:

![image](https://github.com/home-assistant/core/assets/102795312/23eec263-b97a-4303-8a3b-c56e58a6cdbc)

The list is ordered and each index represents the local hour of the day: index 0 = 00:00:00, index 10 = 10:00:00
The profile changes over time depending on the settings of the devices or the user interaction.

I intend to add a custom service to control the profile in a separate pull request where the user can change one or more of the hour temperatures for the device.

Reply:
I see. Instead of this sensor, in a separate PR, add custom services to get and set the schedule.

Reply:
I have removed the sensor.